[BUG] memory_grant_stats_collector error at installation proces

### Component

Full Dashboard

### Performance Monitor Version

2.4.1

### SQL Server Version

MSSQL server 2022 develope

### Windows Version

Windows server 2022 standard

### Describe the Bug

I tried to install Performance monitor full edition and got this error:
[WARN] Validation complete: 45 collectors succeeded, 2 failed
[ERROR]   memory_grant_stats_collector: Error in memory grant stats collector: The current transaction cannot be committed and cannot support operations that write to the log file. Roll back the transaction.
[ERROR]   memory_grant_stats_collector: The current transaction cannot be committed and cannot support operations that write to the log file. Roll back the transaction.
Installation report saved to: C:\Users\sa.ak\PerformanceMonitor_Install_T-ADI-DBMIG_TADIDBMIG_20260329_191130.txt

### Steps to Reproduce

Reproducebly only in my above environment.

### Expected Behavior

Expected behaviour installs Performance monitor without the memory_grant_stats_collector.

### Actual Behavior

I got the following error:
[ERROR]   memory_grant_stats_collector: Error in memory grant stats collector: The current transaction cannot be committed and cannot support operations that write to the log file. Roll back the transaction.
[ERROR]   memory_grant_stats_collector: The current transaction cannot be committed and cannot support operations that write to the log file. Roll back the transaction.

### Error Messages / Log Output

```text

```

### Screenshots

[PerformanceMonitor_Install_T-ADI-DBMIG_TADIDBMIG_20260329_191130.txt](https://github.com/user-attachments/files/26333351/PerformanceMonitor_Install_T-ADI-DBMIG_TADIDBMIG_20260329_191130.txt)

### Additional Context

Installer log in the attachment

--- a/install/05_delta_framework.sql
+++ b/install/05_delta_framework.sql
@@ -1202,38 +1202,43 @@ BEGIN
 
     END TRY
     BEGIN CATCH
+        DECLARE
+            @error_message nvarchar(4000) = ERROR_MESSAGE();
+
         /*
         Only rollback if we started the transaction
         Otherwise let the caller handle it
         */
-        IF  @trancount_at_entry = 0 
+        IF  @trancount_at_entry = 0
         AND @@TRANCOUNT > 0
         BEGIN
             ROLLBACK TRANSACTION;
         END;
 
-        DECLARE
-            @error_message nvarchar(4000) = ERROR_MESSAGE();
-        
         /*
-        Log the error
+        Log the error only if the transaction is not doomed
+        When called inside a caller's transaction that is doomed (XACT_STATE = -1),
+        we cannot write to the log — the caller must rollback first
         */
-        INSERT INTO
-            config.collection_log
-        (
-            collector_name,
-            collection_status,
-            duration_ms,
-            error_message
-        )
-        VALUES
-        (
-            N'calculate_deltas_' + @table_name,
-            N'ERROR',
-            DATEDIFF(MILLISECOND, @start_time, SYSDATETIME()),
-            @error_message
-        );
-        
+        IF XACT_STATE() <> -1
+        BEGIN
+            INSERT INTO
+                config.collection_log
+            (
+                collector_name,
+                collection_status,
+                duration_ms,
+                error_message
+            )
+            VALUES
+            (
+                N'calculate_deltas_' + @table_name,
+                N'ERROR',
+                DATEDIFF(MILLISECOND, @start_time, SYSDATETIME()),
+                @error_message
+            );
+        END;
+
         RAISERROR(N'Error calculating deltas for %s: %s', 16, 1, @table_name, @error_message);
     END CATCH;
 END;

--- a/install/06_ensure_collection_table.sql
+++ b/install/06_ensure_collection_table.sql
@@ -1206,8 +1206,14 @@ BEGIN
     BEGIN CATCH
         SET @error_message = ERROR_MESSAGE();
 
+        IF @@TRANCOUNT > 0
+        BEGIN
+            ROLLBACK;
+        END;
+
         /*
         Log errors to collection log
+        Must happen after rollback to avoid doomed transaction writes
         */
         INSERT INTO
             config.collection_log
@@ -1228,11 +1234,6 @@ BEGIN
             DATEDIFF(MILLISECOND, @start_time, SYSDATETIME()),
             @error_message
         );
-
-        IF @@TRANCOUNT > 0
-        BEGIN
-            ROLLBACK;
-        END;
 
         THROW;
     END CATCH;

